src/uwtools/config/jinja2.py

```diff
@@ -126,17 +128,25 @@ def dereference(val: _ConfigVal, context: dict, local: Optional[dict] = None) ->
     :param val: A value possibly containing Jinja2 syntax.
     :param context: Values to use when rendering Jinja2 syntax.
     :param local: Local sibling values to use if a match is not found in context.
+    :param keys: The dict keys leading to this value.
     :return: The input value, with Jinja2 syntax rendered.
     """
     rendered: _ConfigVal = val  # fall-back value
     if isinstance(val, dict):
-        return {dereference(k, context): dereference(v, context, local=val) for k, v in val.items()}
+        keys = keys or []
+        new = {}
+        for k, v in val.items():
+            if isinstance(v, UWYAMLRemove):
+                _deref_debug("Removing value at", " > ".join([*keys, k]))
+            else:
+                new[dereference(k, context)] = dereference(v, context, local=val, keys=[*keys, k])
+        return new
```

Comment:
When dereferencing, if a `UWYAMLRemove` value is encountered, simply decline to add it to the new `dict` of dereferenced keys and values, and debug-log a message noting that it is being removed.